From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: PureGero <puregero@gmail.com>
Date: Mon, 14 Mar 2022 18:07:45 +1000
Subject: [PATCH] Using simulation distance leads to mobs never despawning, it
 needs to be bigger

This is a uniquely MultiPaper issue as MultiPaper doesn't tick
chunks at all outside of the simulation distance.

diff --git a/src/main/java/net/minecraft/server/level/ChunkMap.java b/src/main/java/net/minecraft/server/level/ChunkMap.java
index 2d301b3f08af39392640d40e4016c85124f27e09..84e12a09fc4c9f140a42acbf349052344fbda494 100644
--- a/src/main/java/net/minecraft/server/level/ChunkMap.java
+++ b/src/main/java/net/minecraft/server/level/ChunkMap.java
@@ -249,7 +249,7 @@ public class ChunkMap extends ChunkStorage implements ChunkHolder.PlayerProvider
         this.playerGeneralAreaMap.add(player, chunkX, chunkZ, GENERAL_AREA_MAP_SQUARE_RADIUS); // Paper - optimise checkDespawn
         // Paper start - per player mob spawning
         if (this.playerMobDistanceMap != null) {
-            this.playerMobDistanceMap.add(player, chunkX, chunkZ, this.distanceManager.getSimulationDistance());
+            this.playerMobDistanceMap.add(player, chunkX, chunkZ, this.distanceManager.getSimulationDistance() + 1); // MultiPaper - using simulation distance leads to mobs never despawning, it needs to be bigger
         }
         // Paper end - per player mob spawning
     }
@@ -290,7 +290,7 @@ public class ChunkMap extends ChunkStorage implements ChunkHolder.PlayerProvider
         this.playerGeneralAreaMap.update(player, chunkX, chunkZ, GENERAL_AREA_MAP_SQUARE_RADIUS); // Paper - optimise checkDespawn
         // Paper start - per player mob spawning
         if (this.playerMobDistanceMap != null) {
-            this.playerMobDistanceMap.update(player, chunkX, chunkZ, this.distanceManager.getSimulationDistance());
+            this.playerMobDistanceMap.update(player, chunkX, chunkZ, this.distanceManager.getSimulationDistance() + 1); // MultiPaper - using simulation distance leads to mobs never despawning, it needs to be bigger
         }
         // Paper end - per player mob spawning
         this.playerChunkManager.updatePlayer(player); // Paper - replace chunk loader
